simple_salesforce/api.py

```diff
@@ -129,7 +126,8 @@ def __init__(
             self.auth_type = "password"
 
             # Pass along the username/password to our login helper
-            self.session_id, self.sf_instance = SalesforceLogin(
+            self._salesforce_login_partial = partial(
```

Comment:
Should be a quick addition but should ensure that it supports the OAuth and Security token login options

- [ ] Pass through OAuth arguments
- [ ] Pass through security token arguements

Reply:
Just saw this! Summer is coming up soon and I'll likely resume the project I was using this for and thus have time to take a look at this. I'll look into it and come back if I have any questions.

Thanks for taking a look!